### PR TITLE
Sms

### PR DESCRIPTION
**Pull request by krzysztofkrzeslak**

### CODE REVIEW

File: src/LiveObjects.cpp [hunk 1]
```diff
@@ -540,10 +591,33 @@ void LiveObjectsNB::messageCallback(int msg)
...
+    obj["carrier"]=carrier;
+  }
+}
+
+void LiveObjectsNB::sendData()
+{
+  if(m_Protocol == MQTT) LiveObjectsBase::sendData();
+  else
+  {
+    if(m_sPayload.length() > 130)
```
Comment (krzysztofkrzeslak) [inline src/LiveObjects.cpp:824]: To check if possible to send multipart sms

File: src/LiveObjects.cpp [hunk 2]
```diff
@@ -658,10 +777,21 @@ void LiveObjectsGSM::messageCallback(int msg)
...
+    addToStringPayload(m_GSMAcces.status() == GSM3_NetworkStatus_t::GSM_READY);
+    addToStringPayload(strength);
+    addToStringPayload(carrier);
+  }
+  else
+  {
+    JsonObject obj = easyDataPayload[JSONVALUE].createNestedObject("networkInfo");
+    obj["connection_status"] = m_GSMAcces.status() == GSM_READY ? "connected":"disconnected";
+    obj["strength"] = m_GSMScanner.getSignalStrength();
+    obj["carrier"]=m_GSMScanner.getCurrentCarrier();
```
Comment (krzysztofkrzeslak) [inline src/LiveObjects.cpp]: to consider creating a universal function for adding data to payload for every protocol(mqtt, sms, lora in future)

File: src/LiveObjects.h [hunk 3]
```diff
@@ -76,14 +77,16 @@ enum LiveObjects_variableType {
...
 };
 
-enum Security
+enum Mode
 {
   NONE
   ,TLS
+  ,BIN_
+  ,TXT
   //,DTLS
```
Comment (krzysztofkrzeslak) [inline src/LiveObjects.h:90]: consider changing "BIN_"  maybe to "BINR" ?

File: src/LiveObjects.h [hunk 4]
```diff
@@ -76,14 +77,16 @@ enum LiveObjects_variableType {
...
   //,LORA
 };
 
-enum Security
+enum Mode
 {
   NONE
   ,TLS
+  ,BIN_
+  ,TXT
```
Comment (krzysztofkrzeslak) [inline src/LiveObjects.h:89]: consider changing "BIN_" maybe to "BINR" ?

**Review by krzysztofkrzeslak — COMMENTED** [review]
To consider merging NB and GSM class

File: src/LiveObjects.h (hunk 4 above)
Comment (mdelain) [reply src/LiveObjects.h:89]: Is it still relevant regarding to the new SMS 'protocol'? Perhaps more accurate to have begin(protocol, debug) with protocol=MQTT, SMS, LORA, and use setSecurity in conjunction with MQTT only (when needed).
+ additional comments on SMS types by email, to come.

**Comment by krzysztofkrzeslak** [comment]
Closing and reopening new with changes only for MKR family boards(Kacp3r3:master)

**Pull request closed without merging**